server/sonar-server/src/main/java/org/sonar/server/es/EsClient.java

```diff
@@ -117,55 +131,60 @@ public void waitForStatus(ClusterHealthStatus status) {
   }
 
   public IndicesExistsRequestBuilder prepareIndicesExist(String... indices) {
-    return new ProxyIndicesExistsRequestBuilder(deprecatedClient.nativeClient(), indices);
+    return new ProxyIndicesExistsRequestBuilder(nativeClient(), indices);
   }
 
   public CreateIndexRequestBuilder prepareCreate(String index) {
-    return new ProxyCreateIndexRequestBuilder(deprecatedClient.nativeClient(), index);
+    return new ProxyCreateIndexRequestBuilder(nativeClient(), index);
   }
 
   public PutMappingRequestBuilder preparePutMapping(String... indices) {
-    return new ProxyPutMappingRequestBuilder(deprecatedClient.nativeClient()).setIndices(indices);
+    return new ProxyPutMappingRequestBuilder(nativeClient()).setIndices(indices);
   }
 
   public SearchRequestBuilder prepareSearch(String... indices) {
-    return new ProxySearchRequestBuilder(deprecatedClient.nativeClient()).setIndices(indices);
+    return new ProxySearchRequestBuilder(nativeClient()).setIndices(indices);
   }
 
   public SearchScrollRequestBuilder prepareSearchScroll(String scrollId) {
-    return new ProxySearchScrollRequestBuilder(scrollId, deprecatedClient.nativeClient());
+    return new ProxySearchScrollRequestBuilder(scrollId, nativeClient());
   }
 
   public GetRequestBuilder prepareGet() {
-    return new ProxyGetRequestBuilder(deprecatedClient.nativeClient());
+    return new ProxyGetRequestBuilder(nativeClient());
   }
 
   public GetRequestBuilder prepareGet(String index, String type, String id) {
-    return new ProxyGetRequestBuilder(deprecatedClient.nativeClient()).setIndex(index).setType(type).setId(id);
+    return new ProxyGetRequestBuilder(nativeClient()).setIndex(index).setType(type).setId(id);
   }
 
   public MultiGetRequestBuilder prepareMultiGet() {
-    return new ProxyMultiGetRequestBuilder(deprecatedClient.nativeClient());
+    return new ProxyMultiGetRequestBuilder(nativeClient());
   }
 
   public CountRequestBuilder prepareCount(String... indices) {
-    return new ProxyCountRequestBuilder(deprecatedClient.nativeClient()).setIndices(indices);
+    return new ProxyCountRequestBuilder(nativeClient()).setIndices(indices);
   }
 
   public BulkRequestBuilder prepareBulk() {
-    return new ProxyBulkRequestBuilder(deprecatedClient.nativeClient());
+    return new ProxyBulkRequestBuilder(nativeClient());
   }
 
   public DeleteRequestBuilder prepareDelete(String index, String type, String id) {
-    return new ProxyDeleteRequestBuilder(deprecatedClient.nativeClient(), index).setType(type).setId(id);
+    return new ProxyDeleteRequestBuilder(nativeClient(), index).setType(type).setId(id);
   }
 
+  /**
+   * @deprecated delete-by-query is dropped from ES 2.0 and should not be used. See
+   * https://www.elastic.co/guide/en/elasticsearch/reference/1.7/docs-delete-by-query.html
+   */
+  @Deprecated
   public DeleteByQueryRequestBuilder prepareDeleteByQuery(String... indices) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)